x-element.js

```diff
@@ -1643,106 +1466,422 @@ class TemplateEngine {
+        update?.updater !== lastUpdate?.updater
+      )
+    ) {
+      // Reset content under certain conditions. E.g., `map(…)` >> `null`.
+      TemplateEngine.#removeBetween(startNode, node);
+      const state = TemplateEngine.#setIfMissing(TemplateEngine.#nodeToArrayState, startNode, () => ({}));
+      TemplateEngine.#clearObject(state);
+    }
+    if (update) {
+      switch (update.updater) {
+        case TemplateEngine.#map:
+          TemplateEngine.#map(node, startNode, update.value, update.identify, update.callback);
+          break;
+        case TemplateEngine.#repeat:
+          TemplateEngine.#repeat(node, startNode, update.value, update.identify, update.callback);
+          break;
+        case TemplateEngine.#unsafe:
+          TemplateEngine.#unsafe(node, startNode, update.value, lastUpdate?.value, update.language);
+          break;
+        case TemplateEngine.#unsafeHTML:
+          TemplateEngine.#unsafeHTML(node, startNode, update.value, lastUpdate?.value);
+          break;
+        case TemplateEngine.#unsafeSVG:
+          TemplateEngine.#unsafeSVG(node, startNode, update.value, lastUpdate?.value);
+          break;
+        default:
+          TemplateEngine.#throwUpdaterError(update.updater, 'content');
+          break;
+      }
+    } else {
+      if (value !== lastValue) {
+        if (TemplateEngine.#symbolToResult.has(value)) {
+          const state = TemplateEngine.#setIfMissing(TemplateEngine.#nodeToArrayState, startNode, () => ({}));
+          const result = TemplateEngine.#symbolToResult.get(value);
+          if (TemplateEngine.#cannotReuseResult(state.result, result)) {
+            TemplateEngine.#removeBetween(startNode, node);
+            TemplateEngine.#clearObject(state);
+            TemplateEngine.#ready(result);
+            TemplateEngine.#commit(result);
+            TemplateEngine.#inject(result, node, { before: true });
+            state.result = result;
+          } else {
+            TemplateEngine.#assign(state.result, result);
+            TemplateEngine.#commit(state.result);
+          }
+        } else if (Array.isArray(value)) {
+          TemplateEngine.#mapInner(node, startNode, null, null, value, 'array');
+        } else {
+          const state = TemplateEngine.#setIfMissing(TemplateEngine.#nodeToArrayState, startNode, () => ({}));
+          if (state.result) {
+            TemplateEngine.#removeBetween(startNode, node);
+            TemplateEngine.#clearObject(state);
+          }
+          const previousSibling = node.previousSibling;
+          if (previousSibling === startNode) {
+            // The `?? ''` is a shortcut for creating a text node and then
+            //  setting its textContent. It’s exactly equivalent to the
+            //  following code, but faster.
+            // const textNode = document.createTextNode('');
+            // textNode.textContent = value;
+            const textNode = document.createTextNode(value ?? '');
```

Comment:
Had to walk back this fork removal. It’s just better performance to do this in one step. I left a fat comment about it though 😅